test/Microsoft.Azure.WebJobs.Extensions.EventHubs.Tests/EventHubEndToEndTests.cs

```diff
@@ -162,6 +182,29 @@ public class EventHubParitionKeyTestJobs
                 }
             }
 
+            /// <summary>
+            /// Test sending partitioned output using EventDataEx extended properties
+            /// </summary>
+            public static async Task SendEvents_EventDataEx_TestHub(
+                string input,
+                [EventHub(TestHubName)] IAsyncCollector<EventData> collector)
+            {
+                byte[] bytes = Encoding.UTF8.GetBytes(input);
+
+                // Send event without PK
+                await collector.AddAsync(new EventData(bytes));
+
+                // Send event with different PKs
+                for (int i = 0; i < 5; i++)
+                {
+                    var evt = new EventDataEx(bytes)
+                    {
+                        PartitionKey = "test_pk" + i
+                    };
+                    await collector.AddAsync(evt);
```

Comment:
This demonstrates the programming model - add an EventDataEx instance to specify partition key.